src/main/java/com/minecolonies/coremod/commands/AbstractCommandParser.java

```diff
@@ -0,0 +1,649 @@
+        if (currentMenuTreeNode.hasChildren())
+        {
+            final StringBuilder sb = new StringBuilder();
+            buildParentPathString(sb, currentMenuTreeNode);
+            sb.append(" <");
+            boolean first = true;
+            for (final TreeNode<IMenu> childTreeNode : currentMenuTreeNode.getChildren())
+            {
+                if (first)
+                {
+                    first = false;
+                }
+                else
+                {
+                    sb.append('|');
+                }
+                sb.append(childTreeNode.getData().getMenuItemName());
+            }
+            sb.append('>');
+            return sb.toString();
+        }
+        else
+        {
+            final StringBuilder sb = new StringBuilder();
+            buildParentPathString(sb, currentMenuTreeNode);
+            final IMenu menu = currentMenuType.getMenu();
+            if (!currentMenuType.isNavigationMenu())
+            {
+                final ActionMenu actionMenu = (ActionMenu) menu;
+                for (final ActionArgument actionArgument : actionMenu.getActionArgumentList())
+                {
+                    sb.append(' ').append(actionArgument.getUsage());
+                }
+            }
+            return sb.toString();
+        }
+    }
+
+    private static void buildParentPathString(final StringBuilder sb, final TreeNode<IMenu> menuTreeNode)
+    {
+        final List<TreeNode<IMenu>> treeNodeMenuListMinusRoot = new ArrayList<>();
+        TreeNode<IMenu> currentMenuTreeNode = menuTreeNode;
+        while (null != currentMenuTreeNode.getParent())
+        {
+            treeNodeMenuListMinusRoot.add(currentMenuTreeNode);
+            currentMenuTreeNode = currentMenuTreeNode.getParent();
+        }
+        sb.append('/').append(currentMenuTreeNode.getData().getMenuItemName());
+
+        Collections.reverse(treeNodeMenuListMinusRoot);
+        for (final TreeNode<IMenu> treeNode : treeNodeMenuListMinusRoot)
+        {
+            sb.append(' ').append(treeNode.getData().getMenuItemName());
+        }
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String[] args) throws CommandException
+    {
+        final PermissionsChecker permissionChecker = getPermissionsChecker();
+        final ModuleContext moduleContext = getModuleContext();
+        execute(server, sender, args, permissionChecker, moduleContext);
+    }
+
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String[] args, 
+            @NotNull final PermissionsChecker permissionsChecker, @NotNull final ModuleContext moduleContext) throws CommandException
+    {
+        final BlockPos pos = null;
+        final ParsingResult parsingResult = getTabCompletionsAndParsingHolders(getRoot(), server, sender, args, pos, moduleContext);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "getTabCompletionsAndParsingHolders()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)